app/Notebook.py

```diff
@@ -0,0 +1,152 @@
+import json
+import os
+
+
+class Notebook(object):
+    notebook_format = None
+    cells = []
+    metadata = None
+    nbformat_minor = None
+
+    def read(self, path_to_file):
+        input_headless, ext = os.path.splitext(path_to_file)
+        if ext == ".ipynb":
+            self.read_notebook(path_to_file)
+        elif ext == ".py":
+            self.read_py(path_to_file)
+
```

Comment:
The read related functions should probably be extracted to helper classes. The Notebook class would become only the representation.